src/Model/Order/Order.php

```diff
@@ -122,20 +171,82 @@ public static function create(
                 'businessName' => $businessName->toNative(),
                 'businessAddress' => $businessAddress->toNative(),
                 'contactPerson' => $contactPerson->toNative(),
-                'hasPaid' => $isPaid->toNative(),
                 'amount' => $amount->toNative(),
+                'paymentStatus' => $paymentStatus->toNative(),
             ])
         );
         return $instance;
     }
 
     private function applyOrderWasCreated(OrderWasCreated $event): void
     {
-        $this->id              = $event->getId();
-        $this->businessName    = $event->getBusinessName();
+        $this->id = $event->getId();
+        $this->businessName = $event->getBusinessName();
         $this->businessAddress = $event->getBusinessAddress();
-        $this->contactPerson   = $event->getContactPerson();
-        $this->isPaid          = $event->getHasPaid();
-        $this->amount          = $event->getAmount();
+        $this->contactPerson = $event->getContactPerson();
+        $this->amount = $event->getAmount();
+        $this->paymentStatus = $event->getPaymentStatus();
+        $this->datePaid = DatePaid::null();
+        $this->paymentAmount = NonNullPaymentAmount::fromNative([
+            'amount' => 0,
+            'currency' => 'gbp'
+        ]);
+        $this->paymentStatus = NonNullPaymentStatus::UNPAID();
     }
+
+    public function pay(): void
+    {
+        $this->recordThat(
+            OrderWasPaid::occur($this->getId()->toNative())
+        );
+    }
+
+    private function applyOrderWasPaid(OrderWasPaid $event): void
+    {
+        $this->id = $event->getId();
+        $this->datePaid = DatePaid::fromNative($event->createdAt()->format(DATE_RFC3339));
+    }
+
+    public function receivePayment(
+        NonNullEntityId $id,
+        NonNullPayeeName $payeeName,
+        NonNullPaymentAmount $paymentAmount,
+        NonNullPaymentType $paymentType
+    ): void {
+        if($paymentAmount->getMoney()->isZero()) {
+            throw new PaymentAmountMustBeGreaterThanZero($paymentAmount);
+        }
+        if($paymentAmount->getMoney()->isNegative()) {
+            throw new PaymentAmountMustNotBeNegative($paymentAmount);
+        }
+        if($this->getPaymentAmount()->getMoney()->add($paymentAmount->getMoney())->greaterThan($this->getAmount()->getMoney())) {
```

Comment:
This if statement could be moved to its own independent method on the aggegate perhaps called: `checkThatAddingAPaymentDoesNotCauseThePaymentTotalToExceedTheTotalOrderAmount(PaymentAmount $amount)` or something like that. This makes the code a bit tidier and makes the intention of the logic easier to comprehend.

Reply:
I've tidied this up and made it a bit shorter.